### PR TITLE
improve color support for themes

### PR DESCRIPTION
**Pull request by mikeferguson**
based on https://github.com/ros-visualization/rviz/pull/1319

### CODE REVIEW

**Comment by mikeferguson** [comment]
This makes dark mode look good on Mac. Before:
2020-08-10-dark-mode


After:
Screen Shot 2020-08-09 at 7 05 25 PM

Also tested against regular mode, all looks good (and this is merged into Melodic)

**Comment by mikeferguson** [comment]
CI appears to be broken due to #589 (seems that the resource_retriever package hasn't been re-released yet with those changes)

**Comment by clalancette** [comment]
> CI appears to be broken due to #589 (seems that the resource_retriever package hasn't been re-released yet with those changes)

Yeah, we need to release a new resource_retriever and a new rviz2.

**Comment by clalancette** [comment]
@ros-pull-request-builder retest this please

**Comment by clalancette** [comment]
@mikeferguson FYI, the current failures look to be style failures with this patch.

**Comment by mikeferguson** [comment]
@clalancette style fixes are fixed - but there's some weird failure I can't seem to figure out:

```
WARNING: The converted file for the input file '/home/jenkins-agent/workspace/Rpr__rviz__ubuntu_focal_amd64/ws/test_results/rviz2/Testing/20200814-1326/Test.xml' doesnt match the JUnit format.
09:26:57 WARNING: At line 2 of file:/home/jenkins-agent/workspace/Rpr__rviz__ubuntu_focal_amd64/generatedJUnitFiles/4c631cba-8659-477a-bd2f-9af6c0c9a750/GoogleTest/TEST-3d6e02d0-4e95-4845-9763-7699c02eb1d7.xml:Premature end of file.
09:26:57 WARNING: Technical validation:Premature end of file.
09:26:57 ERROR: Premature end of file.
09:26:57 ERROR: Premature end of file.
09:26:57 ERROR: Step ‘Publish xUnit test result report’ failed: The converted file for the result file '/home/jenkins-agent/workspace/Rpr__rviz__ubuntu_focal_amd64/ws/test_results/rviz2/Testing/20200814-1326/Test.xml' (during conversion process for the metric 'GoogleTest') is not valid. The report file has been skipped.
```

 - but in the "rviz2" section, there are no failures listed - or even any gtests (and the output for the rviz2 section looks the same as a PR that succeeds: http://build.ros2.org/job/Rpr__rviz__ubuntu_focal_amd64/27/console#console-section-2)

**Comment by clalancette** [comment]
> @clalancette style fixes are fixed - but there's some weird failure I can't seem to figure out:

Yeah, sorry, we are having some infrastructure issues at the moment.  Those should be fixed once https://github.com/ros-infrastructure/ros_buildfarm/pull/828 is merged.  If that is the only failure, then you are good to go.

**Comment by mikeferguson** [comment]
@clalancette -- thanks for the quick feedback -- yeah that appears to be the only issue (I had forgotten to lint before opening the PR - thos issues are fixed).

**Comment by dirk-thomas** [comment]
@ros-pull-request-builder retest this please

**Comment by mikeferguson** [comment]
@clalancette tests are now passing!

**Review by jacobperron — APPROVED** [review]
LGTM
* Linux [Build Status](http://ci.ros2.org/job/ci_linux/12025/)
* Linux-aarch64 [Build Status](https://ci.ros2.org/job/ci_linux-aarch64/7050/)
* macOS [Build Status](http://ci.ros2.org/job/ci_osx/9800/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/11957/)

File: rviz_common/src/rviz_common/properties/property.cpp [hunk 1]
```diff
@@ -251,14 +253,8 @@ void Property::setParent(Property * new_parent)
...
 {
-  if (role == Qt::ForegroundRole &&
-    ( parent_ && parent_->getDisableChildren() ) )
-  {
-#if QT_VERSION < QT_VERSION_CHECK(5, 0, 0)
-    return Qt::gray;
-#else
-    return QColor(Qt::gray);
-#endif
+  if (role == Qt::TextColorRole && parent_ && parent_->getDisableChildren()) {
```
Comment (jacobperron) [inline rviz_common/src/rviz_common/properties/property.cpp]: The Windows build is complaining that `Qt::TextColorRole` is deprecated.
[Docs suggest changing this to `Qt::ForegroundRole`](https://doc.qt.io/qt-5/qt.html#ItemDataRole-enum).

Comment (mikeferguson) [reply rviz_common/src/rviz_common/properties/property.cpp]: Looks right to me on the Mac - want to rerun the other tests?

**Comment by jacobperron** [comment]
Windows: [Build Status](https://ci.ros2.org/job/ci_windows/11962/)

**Comment by jacobperron** [comment]
@mikeferguson thanks!

**Pull request merged**